pkg/pravega/pravega_segmentstore.go

```diff
@@ -245,3 +282,40 @@ func configureTier2Filesystem(podSpec *corev1.PodSpec, pravegaSpec *api.PravegaS
 		})
 	}
 }
+
+func makeSegmentStoreServices(pravegaCluster *api.PravegaCluster) []*corev1.Service {
+	var service *corev1.Service
+	services := make([]*corev1.Service, pravegaCluster.Spec.Pravega.SegmentStoreReplicas)
+
+	for i := int32(0); i < pravegaCluster.Spec.Pravega.SegmentStoreReplicas; i++ {
+		service = &corev1.Service{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Service",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      k8sutil.ServiceNameForSegmentStore(pravegaCluster.Name, i),
+				Namespace: pravegaCluster.Namespace,
+				Labels:    k8sutil.LabelsForSegmentStore(pravegaCluster),
+				OwnerReferences: []metav1.OwnerReference{
+					*k8sutil.AsOwnerRef(pravegaCluster),
+				},
+			},
+			Spec: corev1.ServiceSpec{
+				Type:                  corev1.ServiceTypeNodePort,
```

Comment:
Is there any way of telling the Operator to work WITHOUT a node port?  How does the Segmentstore know what IP address to register with the controller?

Reply:
> Is there any way of telling the Operator to work WITHOUT a node port? 
Not yet. But it would be interesting to add the ability to enable/disable external connectivity to avoid having to create those extra resources.

> How does the Segmentstore know what IP address to register with the controller?
I'm still working on that part. The host IP can be obtained through the Downward API, but not sure yet about the node port exposed for that segmentstore instance...